extensions/script-libraries/dropbox/dropbox.livecodescript

```diff
@@ -0,0 +1,5729 @@
+variable will be the binary data of the file.
+
+If an error occurred it will be a JSON string in the following form:
+
+````
+{
+"error_summary": string,
+"error": {
+".tag": string
+}
+}
+````
+*/
+command dropboxDownload pAccessToken, pPath, pCallback
+   /*
+   DownloadArg
+   path String The path of the file to download.
+   {"path": "/apps/MyApp/Prime_Numbers.txt"}
+   Example: id {"path": "id:a4ayc_80_OEAAAAAAAAAYa"}
+   Example: rev  {"path": "rev:a1c10ce0dd78"}
+   */
+   local tHeader, tPost, tURL
+   put __download_HEADER(pAccessToken,pPath) into tHeader
+   put __download_POST() into tPost
+   put __download_URL() into tUrl
+   
+   __API tHeader, tPost, tURL, pCallback
+   if the result is empty then
+      return it for value
+   else
+      return the result for error
+   end if
+end dropboxDownload
+
+/**
+
+Returns the metadata for a file or folder.
+
+Syntax: dropboxGetMetadata <pAccessToken>, <pPath>, <pIncludeMediaInfo>, [<pCallback>]
+
+Description:
+>*Note:* Metadata for the root folder is unsupported.
+
+If the path is a folder, all its contents will be deleted too.
+A successful response indicates that the file or folder was deleted. The returned
+metadata will be the corresponding FileMetadata or FolderMetadata for the item at
+time of deletion, and not a DeletedMetadata object.
+
+If the callback parameter is not empty the request will be asynchronus and
+when complete the callback will be sent to the object that accessed the API.
+The callback will be sent with three parameters:
+
+- The request ID which will be the value of the it variable after calling the command
+- The HTTP response code
+- The data returned which will be the same data as documented for the it variable in a synchronous request.
+
+
+Parameters:
+pAccessToken: An OAuth2 Access token to access the user's account
+
+pPath: Path in the user's Dropbox to be deleted.
+
+pIncludeMediaInfo (boolean): If true, FileMetadata.media_info is set for photo and
+video. The default for this field is False.
+
+pCallback: The handler to call when the request is complete. If empty the command
+will block until complete. Requires tsNet with LiveCode Indy License.
+
+The result: An error if the HTTP request failed
+
+It: For asynchronous requests it will contain a request ID. This may be used with
+tsNet's tsNetGetStatus function. If the request is synchronous and successful the it
+variable will be a UFT8 encoded JSON string in the form:
+
+````
+{
+".tag": string,
+"name": string,
+"id": string,
+"client_modified": string,
+"server_modified": string,
+"rev": string,
+"size": integer,
+"path_lower": string,
+"path_display": string,
+"sharing_info": {
+"read_only": boolean,
+"parent_shared_folder_id": string,
+"modified_by": string
+},
+"property_groups": [
+{
+"template_id": string,
+"fields": [
+{
+"name": string,
+"value": string
+}
+]
+}
+],
+"has_explicit_shared_members": boolean
+}
```

Comment:
I also suggest moving the JSON sample blocks into the "Description" section of the documentation.  I find that having huge blocks for things like the "It" section makes it hard to get a quick overview of the API without having to do a lot of scrolling.  What do you think?